### ISSUE TITLE
Config newly added RuboCop Cops

### ISSUE THREAD

**Issue by sikachu** [cookpad/global-style-guides#163]
Upon running `rubocop` locally, I'm seeing this warning:

> The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
>  - Layout/SpaceAroundMethodCallOperator (0.82)
>  - Lint/RaiseException (0.81)
>  - Lint/StructNewOverride (0.81)
>  - Style/ExponentialNotation (0.82)
>  - Style/HashEachMethods (0.80)
>  - Style/HashTransformKeys (0.80)
>  - Style/HashTransformValues (0.80)
> For more information: https://docs.rubocop.org/en/latest/versioning/

We probably should look into what should be the values for those cops.

/cc @cookpad/web-chapter

**Comment by brunobrgs** [comment]
## Layout/SpaceAroundMethodCallOperator
https://docs.rubocop.org/en/latest/cops_layout/#layoutspacearoundmethodcalloperator

```ruby
# bad
foo. bar
foo .bar
foo . bar
foo. bar .buzz
foo
  . bar
  . buzz
foo&. bar
foo &.bar
foo &. bar
foo &. bar&. buzz
RuboCop:: Cop
RuboCop:: Cop:: Cop
:: RuboCop::Cop

# good
foo.bar
foo.bar.buzz
foo
  .bar
  .buzz
foo&.bar
foo&.bar&.buzz
RuboCop::Cop
RuboCop::Cop::Cop
::RuboCop::Cop
```

## Lint/RaiseException
https://docs.rubocop.org/en/latest/cops_lint/#lintraiseexception

```ruby
# bad
raise Exception, 'Error message here'

# good
raise StandardError, 'Error message here'
```

## Lint/StructNewOverride
https://docs.rubocop.org/en/latest/cops_lint/#lintstructnewoverride

```ruby
# bad
Bad = Struct.new(:members, :clone, :count)
b = Bad.new([], true, 1)
b.members #=> [] (overriding `Struct#members`)
b.clone #=> true (overriding `Object#clone`)
b.count #=> 1 (overriding `Enumerable#count`)

# good
Good = Struct.new(:id, :name)
g = Good.new(1, "foo")
g.members #=> [:id, :name]
g.clone #=> #<struct Good id=1, name="foo">
g.count #=> 2
```

### Style/ExponentialNotation
https://docs.rubocop.org/en/latest/cops_style/#styleexponentialnotation

```ruby
# Enforces the mantissa to have no decimal part and no
# trailing zeroes.

# bad
3.2e7
0.1e5
120e4

# good
32e6
1e4
12e5

# Enforces a mantissa between 1 (inclusive) and 10 (exclusive).

# bad
10e6
0.3e4
11.7e5
3.14e0

# good
1e7
3e3
1.17e6
3.14
```

### Style/HashEachMethods
https://docs.rubocop.org/en/latest/cops_style/#stylehasheachmethods

```ruby
# bad
hash.keys.each { |k| p k }
hash.values.each { |v| p v }

# good
hash.each_key { |k| p k }
hash.each_value { |v| p v }
```

### Style/HashTransformKeys
https://docs.rubocop.org/en/latest/cops_style/#stylehashtransformkeys

```ruby
# bad
{a: 1, b: 2}.each_with_object({}) { |(k, v), h| h[foo(k)] = v }
{a: 1, b: 2}.map { |k, v| [k.to_s, v] }

# good
{a: 1, b: 2}.transform_keys { |k| foo(k) }
{a: 1, b: 2}.transform_keys { |k| k.to_s }
```

### Style/HashTransformValues
https://docs.rubocop.org/en/latest/cops_style/#stylehashtransformvalues

```ruby
# bad
{a: 1, b: 2}.each_with_object({}) { |(k, v), h| h[k] = foo(v) }
{a: 1, b: 2}.map { |k, v| [k, v * v] }

# good
{a: 1, b: 2}.transform_values { |v| foo(v) }
{a: 1, b: 2}.transform_values { |v| v * v }
```

**Comment by ChrisBr** [comment]
I'm fine with all of the above 👍

**Comment by jesseclark** [comment]
I'm fine with most of @brunobrgs suggestions too. 

There are two options that he showed for Style/ExponentialNotation though. 

The top one is `:integral`
```
EnforcedStyle: integral

# Enforces the mantissa to have no decimal part and no
# trailing zeroes.

# bad
3.2e7
0.1e5
120e4

# good
32e6
1e4
12e5
```

And the second one is `:scientific`:
```
EnforcedStyle: scientific (default)

# Enforces a mantissa between 1 (inclusive) and 10 (exclusive).

# bad
10e6
0.3e4
11.7e5
3.14e0

# good
1e7
3e3
1.17e6
3.14
```

I prefer the `:scientific` style to the `:integral`. 

However, I also can't remember the last time I had to write a number in exponentional notation in Ruby so I'd probably be unaffected by either configuration 😆 .

**Comment by brunobrgs** [comment]
@jesseclark They are just examples from rubocop docs

**Comment by balvig** [comment]
@sikachu how about we try enabling the defaults for all? 
They look good to me (thank you @brunobrgs, that was very helpful 👌)
